### PR TITLE
Wire up serde feature for ars-collections

### LINKED ISSUE #53: epic: Collections
## Goal

Implement the `ars-collections` foundation — Collection trait, Key, Node, Selection model, type-ahead, builder, and advanced collection types (async, virtual, tree, filtered/sorted), mutable wrappers, accessibility announcements, and DnD integration.

## Layer

Subsystem

## Framework

None

## Test tier

Unit

## Spec refs

- `spec/foundation/06-collections.md`

## Decomposition

This epic covers the complete `ars-collections` crate implementation across multiple delivery waves:

### Wave 2 (core types and traits) ✅
- Implement Key, NodeType, and Node (#62)
- Implement Collection trait and StaticCollection (#63)
- Implement selection::Mode, Set, and State (#64)

### Wave 3 (extensions) ✅
- Implement type-ahead / type-select state machine (#70)
- Implement CollectionBuilder (#71)

### Wave 4 (advanced) ✅
- Implement AsyncCollection with pagination support (#81)
- Implement Virtualizer core and vertical layout math (#533)
- Implement advanced layout strategies and directional behavior for Virtualizer (#534)
- Implement TreeCollection (#83)
- Implement FilteredCollection and SortedCollection (#84)
- Implement CollationSupport and CollatorCache (#518)

### Wave 5 (completion)
- Implement MutableListData, MutableTreeData, and CollectionChange (#547)
- Implement CollectionChangeAnnouncement, CollectionMessages, and OnAction (#548)
- Implement DnD collection traits and types (#549)
- Improve edge-case test coverage (#550)

## Context

Waves 2–4 are complete: 464 tests passing, 99.47% line coverage, 6,221 LOC. The core collection types, selection model, navigation, typeahead, async loading, virtualization, tree, filter/sort, and collation are implemented. Wave 5 adds the remaining spec types: mutable collection wrappers with change tracking (§1.8), accessibility announcements (§1.6.1), OnAction callback alias (§3.1), and full DnD collection integration (§10).

See [foundation-completion-roadmap.md](docs/implementation/foundation-completion-roadmap.md) for full task details.

### PR DESCRIPTION
**Pull request by fogodev**
## Summary

- Wires up the previously-declared but unused `serde` feature on `ars-collections` so every public collection type can round-trip through `Serialize`/`Deserialize`.
- Fixes a feature-gating bug where enabling `uuid` alone forced uuid's serde implementation (and therefore serde itself) into the build — now a clean three-combination matrix via `uuid?/serde`.
- Adds `tests/serde.rs` with round-trip coverage for every newly serde-enabled type, anchored on the canonical `Set` test from `spec/testing/13-policies.md` §252–260.
- Syncs `spec/foundation/06-collections.md` with the final Cargo.toml layout and module tree.

### Types now deriving Serialize + Deserialize (behind `serde`)

- `Key` (String / Int / Uuid variants — `Uuid` gated on `uuid` + `serde`)
- `NodeType`, `Node<T>`
- `AsyncLoadingState`
- `CollectionChange<K>`
- `selection::{Mode, Behavior, Set, DisabledBehavior, State}`
- `SortDirection`, `SortDescriptor<K>`

Generic types (`Node<T>`, `CollectionChange<K>`, `SortDescriptor<K>`) carry matching serde bounds on their parameter.

### Cargo.toml feature-gating fix

Before — `uuid` alone transitively pulled serde into the build:
```toml
uuid    = { version = "1", default-features = false, features = ["serde"], optional = true }
serde   = ["dep:serde", "indexmap/serde"]
```

After — weak-dep activation only when both features are on:
```toml
uuid    = { version = "1", default-features = false, optional = true }
serde   = ["dep:serde", "indexmap/serde", "uuid?/serde"]
```

| Features on      | Result                            |
| ---------------- | --------------------------------- |
| `uuid`           | uuid crate, no serde code         |
| `serde`          | serde + indexmap/serde, no uuid   |
| `uuid` + `serde` | uuid crate **with** serde derives |

## Test plan

- [x] `cargo test -p ars-collections --all-features` — new round-trip tests pass (Key variants, NodeType, Node with/without value, every selection enum, SortDirection, SortDescriptor, AsyncLoadingState, all CollectionChange variants)
- [x] `cargo check -p ars-collections` with each feature combination (`--no-default-features`, `+uuid`, `+serde`, `+uuid,+serde`) — all clean
- [x] `cargo xci` — all 13 steps pass (fmt, check, clippy, unit, release, integration, adapter, coverage, feature matrices × 5)
- [x] `spec/foundation/06-collections.md` Cargo.toml example and module tree match the crate's real layout

Closes #53

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/ars-collections/src/sorted_collection/mod.rs [hunk 1]
```diff
@@ -45,6 +46,14 @@ impl Display for SortDirection {
 /// `K` is the column key type — typically the same `Key` used by the
 /// collection, but any type works (e.g., a string column identifier).
 #[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+    feature = "serde",
+    serde(bound(
+        serialize = "K: serde::Serialize",
+        deserialize = "K: serde::de::DeserializeOwned"
```
[the remark this answers is not in the document]
Comment (fogodev) [reply crates/ars-collections/src/sorted_collection/mod.rs]: Fixed in 5841118. Dropped the explicit `#[serde(bound = ...)]` on `SortDescriptor<K>`, `Node<T>`, and `CollectionChange<K>` so serde's default derive generates the permissive `K: Deserialize<'de>` bound instead of the over-strict `DeserializeOwned`. Borrowed generics (`&str` column IDs, etc.) now deserialize as intended.

Added a compile-time regression assertion in `tests/serde.rs` (`BORROWED_GENERIC_PARAMS_DESERIALIZE_ASSERTION`) that exercises `SortDescriptor<&'a str>`, `Node<&'a str>`, and `CollectionChange<&'a str>` with a non-`'static` lifetime — if the bound is ever tightened back, the test file fails to compile.
